### PR TITLE
EPMRPP-37698 || Email notifications backup

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * @author Andrei Varabyeu
+ */
+public class EmailConfigUpdatedEvent extends BeforeEvent<Project> implements ActivityEvent {
+
+	private final ProjectEmailConfigDTO updateProjectEmailRQ;
+	//	private final ProjectEmailConfigDTO updateProjectEmailRQ;
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java]: ?

File: src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	private final Long updatedBy;
+
+	public EmailConfigUpdatedEvent(Project before, ProjectEmailConfigDTO updateProjectEmailRQ, Long updatedBy) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java]: remove a word email

File: src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		activity.setCreatedAt(LocalDateTime.now());
+		activity.setAction(ActivityAction.UPDATE_PROJECT.getValue());
+		activity.setActivityEntityType(Activity.ActivityEntityType.EMAIL_CONFIG);
+		activity.setProjectId(getBefore().getId());
+		activity.setUserId(updatedBy);
+		activity.setObjectId(getBefore().getId());
+
+		ActivityDetails details = new ActivityDetails(getBefore().getName());
+		processEmailConfiguration(details, getBefore(), updateProjectEmailRQ);
+		//		processEmailConfiguration(details, getBefore(), updateProjectEmailRQ);
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java]: ?

File: src/main/java/com/epam/ta/reportportal/core/integration/impl/GetIntegrationHandlerImpl.java [hunk 4]
```diff
@@ -56,4 +67,30 @@ public IntegrationResource getGlobalIntegrationById(Long integrationId) {
 
 		return IntegrationConverter.TO_INTEGRATION_RESOURCE.apply(integration);
 	}
+
+	@Override
+	public Optional<Integration> findEnabledByProjectIdOrGlobalAndIntegrationGroup(Long projectId, IntegrationGroupEnum integrationGroup) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/integration/impl/GetIntegrationHandlerImpl.java]: method that starts from 'find' are methods of the dao layer. use get

File: src/main/java/com/epam/ta/reportportal/core/integration/impl/GetIntegrationHandlerImpl.java [hunk 5]
```diff
@@ -56,4 +67,30 @@ public IntegrationResource getGlobalIntegrationById(Long integrationId) {
...
+
+		if (!CollectionUtils.isEmpty(integrations)) {
+
+			return integrations.stream()
+					.filter(Integration::isEnabled)
+					.findFirst();
+
+		} else {
+
+			return integrationRepository.findAllGlobalInIntegrationTypeIds(integrationTypeIds)
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/integration/impl/GetIntegrationHandlerImpl.java]: what about using private methods for better reading?

File: src/main/java/com/epam/ta/reportportal/core/project/UpdateProjectHandler.java [hunk 6]
```diff
@@ -51,8 +51,8 @@ OperationCompletionRS updateProject(ReportPortalUser.ProjectDetails projectDetai
 	 * @param user            User performing that update
 	 * @return Operation Result
 	 */
-	OperationCompletionRS updateIntegration(ReportPortalUser.ProjectDetails projectDetails, ReportPortalUser user,
-			UpdateIntegrationRQ updateProjectRQ);
+	OperationCompletionRS updateProjectEmailConfig(ReportPortalUser.ProjectDetails projectDetails, ReportPortalUser user,
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/project/UpdateProjectHandler.java]: not email

File: src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java]: removed

File: src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java]: removed

File: src/main/java/com/epam/ta/reportportal/core/integration/impl/GetIntegrationHandlerImpl.java (hunk 5 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/integration/impl/GetIntegrationHandlerImpl.java]: done

File: src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/events/activity/EmailConfigUpdatedEvent.java]: removed

**Pull request merged**